dotnet/src/Extensions/Planning.FlowPlanner/FlowPlanner.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using NS of Plan
+namespace Microsoft.SemanticKernel.Planning;
+#pragma warning restore IDE0130
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.Planning.Flow;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+/// <summary>
+/// A planner that execute plan in iterative way
+/// </summary>
+public class FlowPlanner
+{
+    private readonly KernelBuilder _kernelBuilder;
+
+    private readonly IFlowStatusProvider _flowStatusProvider;
+
+    private readonly Dictionary<object, string?> _globalSkillCollection;
+
+    private readonly IFlowValidator _flowValidator;
+
+    private readonly FlowPlannerConfig? _config;
+
+    /// <summary>
+    /// Initialize a new instance of the <see cref="FlowPlanner"/> class.
+    /// </summary>
+    /// <param name="kernelBuilder">The semantic kernel builder.</param>
+    /// <param name="flowStatusProvider">The flow status provider.</param>
+    /// <param name="globalSkillCollection">The global skill collection</param>
+    /// <param name="validator">The flow validator.</param>
+    /// <param name="config">Optional configuration object</param>
+    public FlowPlanner(
+        KernelBuilder kernelBuilder,
```

Comment:
The idea of having kernel builder is that for each step, we could use an isolated kernel.
Otherwise when we present functions in ReAct steps, unwanted skills would show up there and mislead the reasoning.